File: ProyectoFinal_LP/src/app/Interfaces/Evento.ts
// Interface principal para GET
export interface Evento {
  id?: number;
  titulo?: string;
  descripcion?: string;
  fechaInicio?: string;
  fechaFin?: string;
  aforoMaximo?: number;
  estado?: string;
  localId?: number;
  createdAt?: string;
  updatedAt?: string;

}

// Interface para POST (sin id)
export interface EventoPost {
  titulo: string;
  descripcion: string;
  fechaInicio: string;
  fechaFin: string;
  aforoMaximo: number;
  estado: string;
  localId: number;
}

// Interface para PUT (requiere id)
export interface EventoPut {
  id: number;
  titulo: string;
  descripcion: string;
  fechaInicio: string;
  fechaFin: string;
  aforoMaximo: number;
  estado: string;
  localId: number;
}

File: ProyectoBackend/pizzeriaBackend/src/main/java/com/pizza/pizzeriaBackend/model/Evento.java
package com.pizza.pizzeriaBackend.model;


import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "eventos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Evento {

    public enum EstadoEvento {
        ACTIVO, INACTIVO, CANCELADO
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titulo;
    private String descripcion;
    private LocalDateTime fechaInicio;
    private LocalDateTime fechaFin;
    private Integer aforoMaximo;

    @Enumerated(EnumType.STRING)
    private EstadoEvento estado = EstadoEvento.ACTIVO;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "local_id", nullable = false)
    private Local local;
}